### ISSUE TITLE
job update stanza

### ISSUE THREAD

**Issue by Garagoth** [hashicorp/nomad#3878]
### Nomad version
0.7.1

### Operating system and Environment details
CentOS 7

### Issue
Placing an update stanza on job level and not adding it on group level emits a message "A best effort conversion to new update stanza introduced in v0.6.0 applied. Please update upgrade stanza before v0.7.0."

Taken from  nomad/nomad/structs/structs.go, line 1657 (master branch):
```
// COMPAT: Remove in 0.7.0
	// Rewrite any job that has an update block with pre 0.6.0 syntax.
	jobHasOldUpdate := j.Update.Stagger > 0 && j.Update.MaxParallel > 0
	if jobHasOldUpdate && j.Type != JobTypeBatch {
		// Build an appropriate update block and copy it down to each task group
		base := DefaultUpdateStrategy.Copy()
		base.MaxParallel = j.Update.MaxParallel
		base.MinHealthyTime = j.Update.Stagger
```
Stagger is set by default to 30s and just setting max_parallel will cause jobHasOldUpdate to be set to true.
Then it sets min healthy time to value of stagger, ignoring my value set in min_healthy_time.
Setting stagger to 0 is not allowed.

Maybe just see it there is MinHealthyTime set on j.Update and if it is consider it to be a new syntax?

### Job file (if appropriate)
```
job "test" {
  datacenters = ["dc1", "dc2"]
  update {
    canary = 0
    max_parallel = 1
    health_check  = "checks"
    min_healthy_time = "1m"
    healthy_deadline = "20m"
    auto_revert = true
    stagger = 0
  }
  group "group1" {
    # anything except update element
    count = 2
    task "task1" {
      # ...
    }
  }
}
```

**Comment by preetapan** [comment]
@Garagoth could you try removing stagger from the update stanza above? That should fix things for you. 

We will remove that compatibility block in the next release, that should have been removed in 0.7.0 itself.

**Comment by Garagoth** [comment]
Removing stagger does not change this, even worsens things. It is set by
default to 30s and sets min_healthy_time to that, ignoring my setting.

**Comment by preetapan** [comment]
This was addressed with #3979
